src/components/hooks/useKeyPress.js

```diff
@@ -0,0 +1,42 @@
+import {useState, useEffect} from 'react';
+
+/**
+ * Custom hook that returns wether a key is pressed on an element.
+ *
+ * @param {string} targetKey  Target key to listen for.
+ * @param {RefObject} elementRef  Element on which to listen for event.
+ * @returns {boolean} Wehter or not a key has been pressed.
+ */
+export default function useKeyPress(targetKey, elementRef) {
+  const [keyPressed, setKeyPressed] = useState(false);
+
+  const downHandler = ({key}) => {
+    if (key === targetKey) {
+      setKeyPressed(true);
+    }
+  };
+
+  const upHandler = ({key}) => {
+    if (key === targetKey) {
+      setKeyPressed(false);
+    }
+  };
+
+  useEffect(() => {
+    if (elementRef.current) {
+      elementRef.current.addEventListener('keydown', downHandler);
+      elementRef.current.addEventListener('keyup', upHandler);
+    }
+
+    return () => {
+      if (elementRef.current) {
+        elementRef.current.removeEventListener('keydown', downHandler);
+        // eslint-disable-next-line
+        elementRef.current.removeEventListener('keyup', upHandler);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
this means the effect is only executed once, what if the ref (elementRef.current) is not set yet at that moment?